mesonbuild/build.py

```diff
@@ -1285,17 +1285,7 @@ def process_kwargs(self, kwargs: BuildTargetKeywordArguments) -> None:
                                         (str, bool))
         self.install_mode = kwargs.get('install_mode', None)
         self.install_tag = stringlistify(kwargs.get('install_tag', [None]))
-        extra_files = kwargs.get('extra_files', [])
-        for i in extra_files:
-            # TODO: use an OrderedSet instead of a list?
-            if i in self.extra_files:
-                continue
-            # TODO: this prevents built `File` objects from being used as
-            # extra_files.
-            trial = os.path.join(self.environment.get_source_dir(), i.subdir, i.fname)
```

Comment:
Is this changing? Are we adding support for new things here?

Reply:
Thanks. I was mostly confused since I wasn't sure how to read the TODO. It seemed to be implying that we *want* to change the status eventually and start allowing generated files.

Reply:
I wrote the TODO, and originally I thought this was buggy behavior, but then realized this field is for files being grouped in an IDE, which I would think generally people don't want to edit generated files so this actually makes sense.